utils/rpms/fpm_common.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+# (C) Copyright 2025 Google LLC
+root="$(realpath "$(dirname "$(dirname "$(dirname "${BASH_SOURCE[0]}")")")")"
+. "${root}/utils/sl/setup_local.sh" > /dev/null
+unset tmp
+trap 'if [ -n "${tmp}" ] && [ -z "${DEBUG:-}" ]; then rm -rf "${tmp}"; fi' EXIT
+tmp="$(mktemp -d)"
+
+export install_list=()
+export PACKAGE_TYPE="dir"
+export dbg_list=()
+export EXTERNAL_DEPENDS=()
+export DEPENDS=()
+export EXTRA_OPTS=()
+export FILTER_LIST=()
+isa="$(uname -m)"
+export isa
+export prefix="${PREFIX:-/usr}"
+export bindir="${BINDIR:-/usr/bin}"
+export libdir="${LIBDIR:-/usr/lib64}"
+export includedir="${INCLUDEDIR:-/usr/include}"
+export datadir="${DATADIR:-/usr/share}"
+export sysconfdir="${SYSCONFDIR:-/etc}"
+export sysctldir="${SYSCTLDIR:-/etc/sysctl.d}"
+export unitdir="${UNITDIR:-/usr/lib/systemd/system}"
+export mandir="${MANDIR:-/usr/share/man}"
+daos_version="$(grep "^Version: " "${root}/utils/rpms/daos.spec" | sed 's/^Version: *//')"
+export daos_version
+daos_release="$(grep "^Release: " "${root}/utils/rpms/daos.spec" | \
+  sed 's/^Release: *//' | sed 's/%.*//')"
+export daos_release
+
+export libfabric_version="1.22.0"
+export mercury_version="2.4.0"
+export argobots_version="1.2"
+export pmdk_version="2.1.0"
+export isal_version="2.30.0"
+export isal_crypto_version="2.24.0"
+
+source utils/rpms/package_names.sh
+
+filter_file() {
+  for filter in "${FILTER_LIST[@]}"; do
+    if [[ "${1}" =~ ${filter} ]]; then
+      echo "Filtering ${1}"
+      return 0
+    fi
+  done
+  return 1
+}
+
+expand_directories() {
+  local file
+  local -n expanded=$1; shift
+  local -n expanddirs=$1; shift
+  for file in "$@"; do
+    if filter_file "${file}"; then
+      continue
+    fi
+    if [ -d "${file}" ]; then
+      local tmparray=()
+      readarray -t tmparray <<< "$(ls -1 -d "${file}"/*)"
+      for subfile in "${tmparray[@]}"; do
+        if filter_file "${subfile}"; then
+          continue
+        fi
+        if [ -d "${subfile}" ]; then
+          expanddirs+=("${subfile}")
+        else
+          expanded+=("${subfile}")
+        fi
+      done
+    else
+      expanded+=("${file}")
+    fi
+  done
+}
+
+list_files() {
+  local -n listvar=$1; shift
+  target_dir="${tmp}${TARGET_PATH}"
+  mkdir -p "${target_dir}"
+  # shellcheck disable=SC2068
+  readarray -t tmparray <<< "$(ls -1 -d $@)"
```

Comment:
```suggestion
  readarray -t tmparray <<< "$(ls -1 -d "$@")"
```

Reply:
Doesn't work because we pass in stuff like libdaos.so.* as an argument and this makes it list that rather than honoring the wild card